source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,316 @@
+                    /** Stop program if file formatted incorrectly.
+		             */
+                    std::cout << ""<< std::endl;
+                    throw std::ios_base::failure("Adiabatic boundary file not formatted correctly. " + adiabatic_boundary_file_name + "Make sure you have lat, lon, value with lat. or lon. varying.");
+                  }
+
+                if (std::fabs(latitudes_iso[0] - latitudes_iso[1]) > 1e-9)
+                  {
+                    /** Calculate delta as half the distance between points.
+		             */
+                    delta = std::fabs((0.5)*(latitudes_iso[0] - latitudes_iso[1]));
+                    /** If flag is 0 then longitudes grouped and we calculate delta from latitudes
+		             */
+                    data_flag = 0;
+                  }
+                else
+                  {
+                    /** Calculate delta as half the distance between points.
+		             */
+                    delta = std::fabs((0.5)*(longitudes_iso[0] - longitudes_iso[1]));
+                    /** If flag is 1 then latitudes are grouped and we calculate delta from longitudes
+                     */
+		    data_flag = 1;
+                  }
+                std::cout << ""<< std::endl;
+                std::cout<<"Adiabatic boundary data interaval  delta = "<< delta << std::endl;
+                std::cout << ""<< std::endl;
+                std::cout<<"Resolution of input data in meters is approximately = "<< delta*111*2 << std::endl;
+                std::cout << ""<< std::endl;
+                count++;
+              }
+          }
+
+    /** Calculate the number of unique longitudes or latitudes from the adiabatic boundary file.
+     */
+	double c,d;
+        int count3;
+        count3 = 0;
+        c = 0;
+        d = 0;
+
+        if ( data_flag == 1 )
+          {
+            c = latitudes_iso[0];
+            d = latitudes_iso[1];
+            count3 = 2;
+
+            while (c-d < 1e-9)
+              {
+                c = d;
+                d = latitudes_iso[count3];
+                count3++;
+              }
+          }
+
+        if ( data_flag == 0 )
+          {
+            c = longitudes_iso[0];
+            d = longitudes_iso[1];
+            count3 = 2;
+
+            while (c-d < 1e-9)
+              {
+                c = d;
+                d = longitudes_iso[count3];
+                count3++;
+              }
+           }
+
+        std::cout << ""<< std::endl;
+        std::cout<<"number of unique latitudes or longitudes in Adiabatic boundary file= "<< count3 - 1 << std::endl;
+        number_coords_depth = count3-1;
+        std::cout << ""<< std::endl;
+      }
+
+  }
+}
+
+
+namespace aspect
+{
+  namespace InitialConditions
+  {
+    ASPECT_REGISTER_INITIAL_CONDITIONS(AdiabaticBoundary,
+                                       "adiabatic boundary",
+                                       "In subsection Adiabatic boundary isotherm we define "
```

Comment:
In fact, I think the text that you put into `parameters.tex` might make for a good starting point for what I ask you above.